### ISSUE TITLE
Add some comments to code

### ISSUE THREAD

**Issue by rcegan** [e-zk/sghs#2]
this code is really complicated and the lack of comments makes it hard to follow what's going on.

**Comment by rcegan** [comment]
Bump

**Comment by e-zk** [comment]
Hey @spartan117au upcoming next release will include comments! 🥳 

Next time, add them yourself. L streamer + you fell off + the hood watches Hasan now + ripbozo

**Comment by rcegan** [comment]
Can't wait, thanks!! This project is going to the moon! 🚀🚀🚀🚀

**Comment by e-zk** [comment]
Soon^tm we will integrate this project with the ETH blockchain and add ourselves to the growing DeFi network!

Your mind will be blown by what We^tm have planned for this project. HODL GANG To the moon no kappa 🚀 🪨 😎
